src/SqsPoller/SqsPollerConfiguration.cs

```diff
@@ -1,48 +1,78 @@
+using System;
+using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
-using Amazon;
 using Amazon.SQS;
+using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using SqsPoller.Abstractions;
+using SqsPoller.Abstractions.Extensions;
+using SqsPoller.Resolvers;
 
 namespace SqsPoller
 {
     public static class SqsPollerConfiguration
     {
         public static IServiceCollection AddSqsPoller(
-            this IServiceCollection services, SqsPoolerConfig config, Assembly[] assembliesWithConsumers)
+            this IServiceCollection services,
+            SqsPollerConfig config,
+            params Assembly[] assembliesWithConsumers)
         {
-            services.AddSingleton(config);
-            services.AddSingleton<IConsumerResolver, ConsumerResolver>();
-            services.AddSingleton(provider => new AmazonSQSClient(CreateSqsConfig(config)));
-
-            services.AddTransient<IHostedService, SqsPollerHostedService>();
-            
-            var types = assembliesWithConsumers.SelectMany(assembly => assembly.GetTypes())
-                .Where(type => type.IsClass && typeof(IConsumer).IsAssignableFrom(type))
+            var types = assembliesWithConsumers.SelectMany(x => x.GetTypes())
+                .Where(x => x.IsClass && typeof(IConsumer).IsAssignableFrom(x))
                 .ToArray();
-
-            foreach (var type in types)
-            {
-                services.AddSingleton(typeof(IConsumer), type);
-            }
             
+            services.AddSqsPoller(sc => config, types);
             return services;
         }
 
-        private static AmazonSQSConfig CreateSqsConfig(SqsPoolerConfig config)
+        public static IServiceCollection AddSqsPoller(
+            this IServiceCollection services,
+            Func<IServiceProvider, SqsPollerConfig> configFactory,
```

Comment:
👍 
It became much better with usage of `Func<IServiceProvider, SqsPollerConfig>`